src/main/java/org/terasology/simpleFarming/systems/BushAuthoritySystem.java

```diff
@@ -0,0 +1,262 @@
+                bushComponent.stages[bushComponent.currentStage].minTime,
+                bushComponent.stages[bushComponent.currentStage].maxTime);
+    }
+
+    /**
+     * Called when an attempt to harvest the bush is made.
+     * If the bush is
+     *
+     * @param event  The activation event
+     * @param entity The entity doing the harvesting
+     */
+    @ReceiveEvent
+    public void onHarvest(ActivateEvent event, EntityRef entity) {
+        EntityRef target = event.getTarget();
+        EntityRef harvester = entity.equals(target) ? event.getInstigator() : entity;
+        /* Ensure the target is a plant and the entities are valid */
+        if (!event.isConsumed() && target.exists() && harvester.exists()
+                && target.hasComponent(BushDefinitionComponent.class)) {
+            BushDefinitionComponent bushComponent = target.getComponent(BushDefinitionComponent.class);
+
+            /* Produce is only given in the final stage */
+            if (bushComponent.currentStage == bushComponent.stages.length - 1) {
+                EntityRef produce;
+                /* Handle a dodgy produce */
+                try {
+                    produce = entityManager.create(bushComponent.produce);
+                    boolean giveSuccess = inventoryManager.giveItem(harvester, target, produce);
+                    if (!giveSuccess) {
+                        Vector3f position = event.getTargetLocation().add(0, 0.5f, 0);
+                        produce.send(new DropItemEvent(position));
+                        produce.send(new ImpulseEvent(random.nextVector3f(15.0f)));
+                    }
+                } catch (NullPointerException e) {
+                    logger.error("Unable to create produce: " + bushComponent.produce);
+                }
+                if (bushComponent.sustainable) {
+                    doBushGrowth(bushComponent, -1);
+                } else {
+                    entity.send(new DoDestroyPlant());
+                    worldProvider.setBlock(bushComponent.position, blockManager.getBlock(BlockManager.AIR_ID));
+                }
+                event.consume();
+            }
+        }
+    }
+
+    /**
+     * Handles the seed drop on plant destroyed event.
+     *
+     * @param event  The event corresponding to the plant destroy.
+     * @param entity Reference to the plant entity.
+     */
+    @ReceiveEvent
+    public void onBushDestroyed(CreateBlockDropsEvent event, EntityRef entity, BushDefinitionComponent bushComponent) {
+        entity.send(new DoDestroyPlant());
+        event.consume();
+    }
+
+    /**
+     * Grow the bush by one stage in a specified direction
+     *
+     * @param bushComponent The definition of the bush to grow
+     * @param direction     The direction to grow in. positive indicates forward & negative indicated backwards
+     */
+    private EntityRef doBushGrowth(BushDefinitionComponent bushComponent, int direction) {
```

Comment:
The term direction is already used for a 3d-direction. I would rename direction to offset / stageOffset

Reply:
Whilst direction can be used for a 3-d direction there isn't an instance of it being used like that in this module so I don't think it's confusing.